src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -468,4 +471,21 @@ private static Channel createDefaultChannel(@NotNull String channelName) {
     private boolean allowDuplicateChannelName(Channel channel) {
         return channel.getExercise() != null || channel.getLecture() != null || channel.getExam() != null;
     }
+
+    /**
+     * Generates the channel name based on the associated lecture/exercise/exam title and a corresponding prefix.
+     * It replaces alternating/consecutive occurrences of spaces and hyphens and limits length of the name to 30 characters.
```

Comment:
I would add that it only allows letters, digits and hyphens and replaces all other characters with hyhens.